backend/LexBoxApi/Services/UserService.cs

```diff
@@ -16,6 +16,34 @@ await dbContext.Users.Where(u => u.Id == id)
             .ExecuteUpdateAsync(c => c.SetProperty(u => u.LastActive, DateTimeOffset.UtcNow));
     }
 
+    public async Task UpdatePasswordStrength(Guid id, LexCore.LoginRequest loginRequest)
+    {
+        var canCalculateScore = !loginRequest.PreHashedPassword;
+        var hasScore = loginRequest.PasswordStrength != null;
+        if ((canCalculateScore || hasScore) &&
+            await dbContext.Users.AnyAsync(u => u.Id == id && u.PasswordStrength == null))
```

Comment:
instead of querying here, just update the Where on the update below to only update where strength is null.

Reply:
I was trying to avoid calling EvaluatePassword when we don't need to, figuring a single DB request would be cheaper than calculating the password strength. The way I wrote it, we only calculate the password strength if the strength in the DB is null. If we switch to only updating when strength is null, we've calculated password strength on every single login, even though we don't need it most of the time. (We only need it once per user, ever: when they reset their password, it'll be calculated by the frontend).

I should measure the performance of doing it either way, and see whether the DB query or the password-strength calculation is faster. But for now, I'm going to leave this one as-is unless you're certain that the DB query is more expensive than the password-strength calculation.

Reply:
I was surprised to discover that it likely is faster to hit the DB than calculate the password strength.
3 DB examples:
![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/f89a9d10-e443-4fe2-b3a4-bb7677905745)

Calculating password strength [according to the docs](https://github.com/dropbox/zxcvbn?tab=readme-ov-file#runtime-latency):
> ~5-20ms for ~25 char passwords on modern browsers/CPUs, ~100ms for passwords around 100 characters

And yet...considering these numbers my opinion is that we don't need to think about performance. I'd prefer simple code. Something like this:
```c#
public async Task UpdatePasswordStrength(Guid id, LexCore.LoginRequest loginRequest)
{
    var score = Clamp(
        loginRequest.PasswordStrength ??
        (loginRequest.PreHashedPassword ? null : Zxcvbn.Core.EvaluatePassword(loginRequest.Password).Score));
    if (score is not null)
    {
        await dbContext.Users.Where(u => u.Id == id)
            .ExecuteUpdateAsync(c => c.SetProperty(u => u.PasswordStrength, score));
    }
}
```

If this gets hit 20 times during a Send/Receive (my memory/understanding is failing me, but I feel like there might be a lot of requests during a Send/Receive 🤔. I also don't know if we get plain-text passwords there), that's gonna cost at most about 2s, which is fairly negligible considering Send/Receive takes so long 😿. But 2s is not nothing.

Reply:
Tims comment made me realize, this code actually is only called once per S&R because `LexProxyService.Login` where this is called from, is called behind a cache in `BasicAuthHandler`, so this is basically just executed once so I'm much less worried about the performance. Do whatever you feel is best Robin, let's get this merged.

Reply:
Then if you're ready to get this merged, could you (both @hahn-kev and @myieye ) re-review with an approval? I'd prefer not to dismiss "change requested" reviews unless there's a good reason to.

Anyway, my decision here is to leave the DB query in place since it's measurably faster than running the password-strength evaluation every time. So I believe that's everything addressed now.